cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -201,4 +220,19 @@ public void executeDataOp(HttpRequest request, HttpResponder responder, @PathPar
     // todo: execute data operation
     responder.sendStatus(HttpResponseStatus.NOT_IMPLEMENTED);
   }
+
+  private Map<String, String> getDatasetProperties(HttpRequest request) {
+    ChannelBuffer content = request.getContent();
+
+    if (content == ChannelBuffers.EMPTY_BUFFER) {
+      return new HashMap<>();
+    }
+
+    Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()), Charsets.UTF_8);
```

Comment:
Use the `try (resource)` syntax instead of `finally`.

Reply:
done.